runtime/compiler/ilgen/Walker.cpp

```diff
@@ -5643,6 +5658,16 @@ TR_J9ByteCodeIlGenerator::loadStatic(int32_t cpIndex)
          genTreeTop(treeTopNode);
          }
 
+      static char *disableFinalFieldFoldingInILGen = feGetEnv("TR_DisableFinalFieldFoldingInILGen");
+      static char *disableStaticFinalFieldFoldingInILGen = feGetEnv("TR_DisableStaticFinalFieldFoldingInILGen");
+      if (!disableFinalFieldFoldingInILGen &&
+          !disableStaticFinalFieldFoldingInILGen &&
+          symbol->isFinal() &&
+          TR::TransformUtil::canFoldStaticFinalField(comp(), load) == TR_yes)
+         {
+         TR::TransformUtil::foldReliableStaticFinalField(comp(), load);
```

Comment:
this will just be based on a whitelist of acceptable fields right?

Reply:
Yes, unless `-Xjit:unleashStaticFieldFolding` is used or some env var is on to turn on folding blindly, which is usually used in performance experiment